src/Symfony/Bridge/PhpUnit/Tests/DeprecationErrorHandler/ConfigurationTest.php

```diff
@@ -391,6 +391,73 @@ public function testBaselineFileWriteError()
         $configuration->writeBaseline();
     }
 
+    public function testExistingIgnoreFile(): void
+    {
+        $filename = $this->createFile();
+        $ignorePatterns = [
+            '/Test message .*/',
+            '/^\d* occurrences/',
+        ];
+        file_put_contents($filename, json_encode($ignorePatterns));
+
+        $configuration = Configuration::fromUrlEncodedString('ignoreFile='.urlencode($filename));
+        $trace = debug_backtrace();
+        $this->assertTrue($configuration->isIgnoredDeprecation(new Deprecation('Test message 1', $trace, '')));
+        $this->assertTrue($configuration->isIgnoredDeprecation(new Deprecation('Test message 2', $trace, '')));
+        $this->assertFalse($configuration->isIgnoredDeprecation(new Deprecation('Test mexxage 3', $trace, '')));
+        $this->assertTrue($configuration->isIgnoredDeprecation(new Deprecation('1 occurrences', $trace, '')));
+        $this->assertTrue($configuration->isIgnoredDeprecation(new Deprecation('1200 occurrences and more', $trace, '')));
+        $this->assertFalse($configuration->isIgnoredDeprecation(new Deprecation('Many occurrences', $trace, '')));
+    }
+
+    public function testIgnoreFilePatternWarning(): void
+    {
+        $filename = $this->createFile();
+        $ignorePatterns = [
+            '/Test message (.*/',
+        ];
+        file_put_contents($filename, json_encode($ignorePatterns));
+
+        $configuration = Configuration::fromUrlEncodedString('ignoreFile='.urlencode($filename));
+        $trace = debug_backtrace();
+        $this->expectWarning();
+        $this->expectWarningMessageMatches('/[Cc]ompilation failed/');
```

Comment:
So if the compilation fails, we get a warning instead of an exception :thinking: ? Is that a limitation of PHP?

Reply:
@greg0ire yes, see https://www.php.net/manual/en/function.preg-filter.php

> If the regex pattern passed does not compile to a valid regex, an E_WARNING is emitted.

However the point is correct, like this it will be impossible to understand what's failing. Changed the code to check individually the patterns during class construction and throwing a RuntimeException with the details of the failing pattern.